ui/apps/platform/src/Components/CompoundSearchFilter/components/AttributeSelector.tsx

```diff
@@ -14,7 +15,7 @@ export type AttributeSelectorProps = {
     selectedEntity: SelectedEntity;
     selectedAttribute: SelectedAttribute;
     onChange: AttributeSelectorOnChange;
-    config: Partial<CompoundSearchFilterConfig>;
+    config: DeepPartial<CompoundSearchFilterConfig>;
```

Comment:
Nit: do we really need to `DeepPartial` the entire object here? If this is to make attribute key/values optional, is there a concise way to do that without making top level `displayName` and `searchCategory`, and all attribute object _values_ optional as well?

Reply:
You are right. I overlooked that. That's a good point. So we want to make the keys of the `CompoundSearchFilterConfig` partial. And then for each value, we want to make the `attributes` partial. Something like this would probably be more appropriate:

```
config: PartialNestedKeys<CompoundSearchFilterConfig, ["attributes"]>
```

I wonder if I could do something like this 🤔

Reply:
@dvail you mentioned a suggestion for how to organize the types on top of Mark's suggestion of relocating the types to the Containers directory. I'll make a note of it below and we can come back to it:



If we got rid of all the specifics, I think we could reduce that types file to something like this (untested/experimental):
```export type BaseSearchFilterAttribute = {
    displayName: string;
    filterChipLabel: string;
    searchTerm: string;
};

export type CoreSearchFilterAttribute = BaseSearchFilterAttribute & {
    inputType: 'autocomplete' | 'text' | 'condition-number' | 'date-picker';
};

export type SelectSearchFilterAttribute = BaseSearchFilterAttribute & {
    inputType: 'select';
    inputProps: {
        options: readonly { label: string; value: string }[];
    };
};

export type SearchFilterAttribute = CoreSearchFilterAttribute | SelectSearchFilterAttribute;

export type SearchFilterConfig = {
    displayName: string;
    searchCategory: string;
    attributes: Partial<Record<string, SearchFilterAttribute>>;
};

export type CompoundSearchFilterConfig = Partial<Record<string, SearchFilterConfig>>;
```

The downside of this being that each Container would be free to create its own types, which may or may not be valid search filters. The tradeoff coming from the use of string everywhere instead of the exact categories/search fields/etc.